vignettes/covariance_fixed_effect.Rmd

```diff
@@ -0,0 +1,85 @@
+---
+title: "Covariance of Fixed Effects in `mmrm`"
+package: mmrm
+bibliography: '`r system.file("REFERENCES.bib", package = "mmrm")`'
+output:
+  rmarkdown::html_document:
+          theme: "spacelab"
+          highlight: "kate"
+          toc: true
+          toc_float: true
+vignette: |
+  %\VignetteIndexEntry{Covariance Structures in `mmrm`}
+  %\VignetteEncoding{UTF-8}
+  %\VignetteEngine{knitr::rmarkdown}
+editor_options:
+  chunk_output_type: console
+  markdown:
+    wrap: 72
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+Here we describe the covariance matrix for fixed effects.
+
+# Introduction
+
+To estimate the covariance matrix of fixed effects, there are many ways.
+In `mmrm` package, we implemented asymptotic, empirical, Jackknife and Kenward-Roger methods.
+For simplicity, the following derivation are all for unweighted mmrm.
+For weighted mmrm, we can follow the [details of weighted leaset square estimator](https://openpharma.github.io/mmrm/main/articles/algorithm.html#weighted-least-squares-estimator).
+
+## Asymptotic Covariance
+
+Asymptotic covariance are derived based on the estimate of $\beta$.
+
+Following the definition in [details in model fitting](algorithm.html#linear-model), we have
+
+\[
+  \hat\beta = (X^\top W X)^{-1} X^\top W Y
+\]
+
+\[
+  cov(\hat\beta) = (X^\top W X)^{-1} X^\top W cov(\epsilon) W X (X^\top W X)^{-1} = (X^\top W X)^{-1}
+\]
+
+Where $W$ is the block diagonal matrix of inverse of covariance matrix of $\epsilon$.
+
+## Empirical Covariance
+
+Empirical covariance, also known as the robust sandwich estimator, is derived by replacing the covariance matrix of $\epsilon$ by observed
+covariance matrix.
+
+\[
+  cov(\hat\beta) = (X^\top W X)^{-1}(\sum_{i}{X_i^\top W_i \hat\epsilon_i\hat\epsilon_i^\top W_i X_i})(X^\top W X)^{-1}
```

Comment:
It would be helpful for maintenance to write this out further into subjects etc. so that the C++ code can better be read by first understanding the formulas here

Reply:
it is by subjects ; the meat part,  $(\sum_{i}{X_i^\top W_i \hat\epsilon_i\hat\epsilon_i^\top W_i X_i})$ is added by subject, and the bread part, $(X^\top W X)^{-1}$ are already calculated as it is the asymptotic covariance

Reply:
ah yeah now I see it thanks :-) ok basically just always good to check / make sure that going from the math in the vignette to the code in C++ is easy, i.e. does not require any reformulations of the math but just implementing it.